src/site/content/en/blog/decrease-frontend-size/index.md

```diff
@@ -0,0 +1,778 @@
+an url of that file. However, if we specify the `limit` option, it will encode files smaller than
+this limit as [a Base64 data url](https://css-tricks.com/data-uris/) and return this url. This
+inlines the image into the JavaScript code and saves an HTTP request:
+
+```js
+// webpack.config.js
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.(jpe?g|png|gif)$/,
+        loader: 'url-loader',
+        options: {
+          // Inline files smaller than 10 kB (10240 bytes)
+          limit: 10 * 1024,
+        },
+      },
+    ],
+  }
+};
+```
+
+```js
+// index.js
+import imageUrl from './image.png';
+// → If image.png is smaller than 10 kB, `imageUrl` will include
+// the encoded image: 'data:image/png;base64,iVBORw0KGg…'
+// → If image.png is larger than 10 kB, the loader will create a new file,
+// and `imageUrl` will include its url: `/2fcd56a1920be.png`
+```
+
+{% Aside %}
+Inlined images reduce the number of separate requests, which is good ([even with
+HTTP/2](https://blog.octo.com/en/http2-arrives-but-sprite-sets-aint-no-dead/)), but increase the
+download/parse time of your bundle and memory consumption. Make sure to not embed large images or a
+lot of them – or increased bundle time would outweigh the benefit of inlining.
+{% endAside %}
+
+
+[`svg-url-loader`](https://github.com/bhovhannes/svg-url-loader) works just like `url-loader` –
+except that it encodes files with the [URL
+encoding](https://developer.mozilla.org/docs/Glossary/percent-encoding) instead of the Base64
+one. This is useful for SVG images – because SVG files are just a plain text, this encoding is
+more size-effective.
+
+```js
+module.exports = {
+    module: {
+    rules: [
+        {
+        test: /\.svg$/,
+        loader: "svg-url-loader",
+        options: {
+            limit: 10 * 1024,
+            noquotes: true,
+        },
+        },
+    ],
+    },
+};
+```
+
+{% Aside %}
+svg-url-loader has options that improve Internet Explorer support, but worsen inlining for
+other browsers. If you need to support this browser, [apply the `iesafe: true`
+option](https://github.com/bhovhannes/svg-url-loader#iesafe).
+{% endAside %}
+
+[`image-webpack-loader`](https://github.com/tcoopman/image-webpack-loader) compresses images that go
+through it. It supports JPG, PNG, GIF and SVG images, so we’re going to use it for all these types.
+
+This loader doesn’t embed images into the app, so it must work in pair with `url-loader` and
+`svg-url-loader`. To avoid copy-pasting it into both rules (one for JPG/PNG/GIF images, and another
+one for SVG ones), we’ll include this loader as a separate rule with [`enforce: 'pre'`](https://webpack.js.org/configuration/module/#rule-enforce):
+
+```js
+// webpack.config.js
+module.exports = {
+module: {
+    rules: [
+    {
+        test: /\.(jpe?g|png|gif|svg)$/,
+        loader: 'image-webpack-loader',
+        // This will apply the loader before the other ones
+        enforce: 'pre',
+    },
+    ],
+},
+};
```

Comment:
```suggestion
// webpack.config.js
module.exports = {
  module: {
    rules: [
      {
        test: /\.(jpe?g|png|gif|svg)$/,
        loader: 'image-webpack-loader',
        // This will apply the loader before the other ones
        enforce: 'pre'
      }
    ]
  }
};
```